Add dark mode to the documentation website

kolchfa-aws opened the pull request:
Addressing multiple requests, adding dark mode to the doc site.

![dark-mode](https://github.com/user-attachments/assets/ef187817-b000-4360-b7ce-cb6d3f353be4)


### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

ashwin-pc commented:
This does work, but given that this PR is meant to only enable dark mode, a lot of the changes here are actually not necessary and not the best way to do it. 

E.g. you make a lot of changes to the colors but changing them from using a generic value like $primary-xxx to a specific color value like $blue-300 and then change what blue -300 means in dark mode.

I don't know too much of this repo, but usually when we do dark/light mode, we keep the color variables constant since they have a specific meaning and update $primary-xxx to use either $blue-300 or $blue-100 for light and dark mode. That way we never go off brand by using color values that aren't a part of the brand and make sure all the primary color ($blue-100 etc) values are part of the brand colors

There are also a lot of other changes that don't seem to be related to dark mode. Also not an issue if they are intended, but it's better to make sure that all the changes in a PR are related to the feature being reviewed and nothing else.

kolchfa-aws commented:
@ashwin-pc Thank you for reviewing! This PR also includes a refactoring of existing code. The custom.scss file had a lot of duplication of the same styles that I extracted into variables. I also extracted button styling so all buttons can use the same style. The custom.scss was hard to follow and extend in its previous state.

kolchfa-aws commented:
@ashwin-pc I addressed your comments - the variables are now semantically named and have different values depending on the mode.

ashwin-pc commented on _sass/_navigation-header.scss:
```diff
@@ -358,31 +360,31 @@ $max-container: 1440px;
                                 @media screen and (max-width: $header-banner-mobile-breakpoint) {
                                     @include body-text;
-                                    color: $primary-open-sky-t3;
+                                    color: #FFFFFF;
```
This will be an issue in light mode right?

kolchfa-aws replied:
This is the default color that's overridden later but I've changed it to use the variable - thanks.

ashwin-pc commented on assets/js/_version-selector.js:
```diff
@@ -138,7 +138,7 @@ const tpl = `
         border-bottom: 1px solid #eee;
         text-decoration: none;
-        color: var(--link-color);
+        color: #0053b8;
```
we dont need this color to be different between light and dark mode?

ashwin-pc reviewed (approved):
Looks good to me. Just confirm that the variables used in some places that have static values is intentional.
